crates/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -958,7 +958,7 @@ fn check_param_type<E: fmt::Debug, S: StorageView<E>, Mode: ExecutionMode>(
                     msg,
                 ));
             };
-            special_argument_validate(bytes, idx as u16, layout)?;
+            bcs_argument_validate(bytes, idx as u16, layout)?;
```

Comment:
We now always check the BCS layout for arguments of `MakeMoveVec` and MoveCall, even before we reach the VM. 
While this isn't strictly necessary, it will make things a lot more predictable for users.